File: search/linear_search/linear_search.py
# coding: utf-8
"""
線形探索
先頭から対象の値を探す。
計算量はO(n)
"""
import random


def linear_search(array, target):
    """
    線形探索
    """
    for v in array:
        if target == v:
            return True
    return False


def main():
    random.seed(3)  # randomシードを指定
    random_array = [random.randint(0, 99) for _ in range(20)]
    print(random_array)
    print("-----LINEAR SEARCH-----")
    print(linear_search(random_array, 77))  # 存在する数
    print(linear_search(random_array, 2))  # 存在しない数


if __name__ == '__main__':
    main()

File: search/linear_search/linearSearch.hs
linearSearch :: [Int] -> Int -> Int
linearSearch xs target =
  head
    [ i
      | i <- [0 .. length xs - 1],
        xs !! i == target
    ]

main :: IO ()
main = do
  let randomList = [30, 75, 69, 16, 47, 77, 60, 80, 74, 8, 77, 1, 60, 33, 70, 29, 24, 91, 60, 69]
  print randomList
  let target = 77
  print $ linearSearch randomList 77